app/js/fm-api/factories/PlayerTransportResource.js

```diff
@@ -21,6 +21,25 @@ angular.module("sn.fm.api").factory("PlayerTransportResource", [
             // Hash with declaration of custom action that should
             // extend the default set of resource actions
             {
+                get: {
+                    method: "GET",
+                    interceptor: {
+                        response: function(response){
+
+                            response.data = {
+                                track: response.data,
+                                status: response.status,
+                                paused: response.headers("Paused")
+                            };
+
+                            response.resource.track = response.data;
+                            response.resource.status = response.status;
+                            response.resource.paused = response.headers("Paused");
+
+                            return response;
+                        }
+                    }
+                },
                 pause: {
```

Comment:
I would recommend doing any logic in the controller, the api should ideally just return us the data as it comes from the API incase we need to use it elsewhere in our app.